docs/specs/user-profile/design.md

```diff
@@ -0,0 +1,331 @@
+
+## Standard attributes
+
+Standard attributes are [OIDC standard claims](https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims).
+All standard attributes are optional.
+
+## Access control on standard attributes
+
+The Admin API and the portal always have full access to all standard attributes.
+
+```yaml
+user_profile:
+  standard_attributes:
+    access_control:
+    - pointer: /given_name
+      access_control: hidden
+    - pointer: /zoneinfo
+      access_control: readonly
+```
+
+Possible values of `access_control` are:
+
+- `hidden`: The standard attribute is hidden from the end-user, the User Info endpoint and the resolver.
+- `internal`: The standard attribute is hidden in the settings page. But it is visible to the User Info endpoint and the resolver.
+- `readonly`: The standard attribute is visible to the end-user, the User Info endpoint and the resolver. The end-user can view but cannot edit it in the settings page.
+- `readwrite`: The standard attribute is visible to the end-user, the User Info endpoint and the resolver. The end-user can view and edit it in the settings page.
+
+The default value of `access_control` of all standard attributes is `readwrite`.
+
+## Standard attributes population
+
+Most of the standard attributes are subject to population.
+
+```yaml
+user_profile:
+  standard_attributes:
+    population:
+      strategy: on_signup
+```
+
+Possible values of `strategy` are
+
+- `none`: No population.
+- `on_signup`: Populate from the identity being used in sign up.
+
+## List of standard attributes subject to population
+
+- `name`
+- `given_name`
+- `family_name`
+- `middle_name`
+- `nickname`
+- `profile`
+- `picture`
+- `website`
+- `gender`
+- `birthdate`
+- `zoneinfo`
+- `locale`
+- `address.formatted`
+- `address.street_address`
+- `address.locality`
+- `address.region`
+- `address.postal_code`
+- `address.country`
+
+### Single-line string standard attributes
+
+- `name`
+- `given_name`
+- `family_name`
+- `middle_name`
+- `nickname`
+- `address.locality`
+- `address.region`
+- `address.postal_code`
+
+### Multi-line string standard attributes
+
+- `address.formatted`
+- `address.street_address`
+
+### URL standard attributes
+
+- `profile`
+- `picture`
+- `website`
+
+### gender
+
+The predefined values are `male` and `female`. It is possible to use other values.
+
+### zoneinfo
+
+The preferred tz database name, e.g. `Asia/Hong_Kong`.
+The list of tz database names are constants, so the input control is a dropdown.
+
+### locale
+
+The preferred BCP47 tag, e.g. `zh-HK`.
+The input control is a dropdown, and the options are the supported languages of the project.
+When it is populated, the supported languages are taken into account.
+
+### birthdate
+
+The supported format is `YYYY-MM-DD`, representing the birthdate, e.g. 1 Jan 1992.
+
+> The OIDC spec also allows `0000-MM-DD` for birthday and `YYYY` for year of birth.
+> We only support `YYYY-MM-DD` for simplicity.
```

Comment:
How much more complex will it be to support `0000-MM-DD` and `YYYY`?

Reply:
The complexity is we do not know the need of the developer. We have to add configuration to let the developer tell us which format(s) they want their end-users to enter.

And we have to answer questions like:

- What if we receive a value from an external identity provider that is not in one of the allowed format.
- How do we present the UI to let the end-user know they can provide their birthdate / birthday / year of birth?

Reply:
I was thinking what if our UI always support YYYY-MM-DD only. But also allow `0000-MM-DD` and `YYYY` in Admin API.